tests/test_browser.py

```diff
@@ -81,20 +80,25 @@ def test__request(httpbin):
         "Content-Type"]
 
 
-def test__request_file(httpbin):
+@pytest.mark.parametrize('expected_content, set_value', [
+    (":-)", True),
+    ("", False)
+])
```

Comment:
On overall, this is good, but this doesn't test the filename. Unfortunately, httpbin doesn't give us this information, so we can't rely on it, but it would be nice to check that the request we send does contain `filename="..."` somewhere. It should be doable by asking the request library the content of the request before sending it to the network. http://docs.python-requests.org/en/master/api/#requests.PreparedRequest should help.

Reply:
Understood, will look into it!